### DISCUSSION TITLE
Lesson 11: I cant get my FlipMoodNft in my Interactions.s.sol to work

### DISCUSSION THREAD

**Discussion by shittuisrael004** [Cyfrin/foundry-full-course-cu#787] [Q&A]
For the tasks, I was able to write the interactions code for MintMoodNft and it worked perfectly, while my FlipMoodNft interactions also run successfully without any error but it does not really change the mood whenever i import it on my anvil localhost network on metamask

Here is my FlipMoodNft interactions code
```solidity
contract FlipMoodNft is Script {

    function run() external {
        address mostRecentlyDeployed = DevOpsTools.get_most_recent_deployment(
            "MoodNft",
            block.chainid
        );
        flipMoodNftOnContract(mostRecentlyDeployed, 1);
    }

    function flipMoodNftOnContract(address contractAddress, uint256 tokenId) public {
        vm.startBroadcast();
        MoodNft(contractAddress).flipMood(tokenId);
        vm.stopBroadcast();
    }

}
```

And even when i try to hardcode the mintNft function to always mint the Sad SVG from start, i still get the Happy SVG whenever I import on my localhost metamask

```solidity
function mintNft() public {
        _safeMint(msg.sender, s_tokenCounter);
        s_tokenIdToMood[s_tokenCounter] = Mood.SAD;
        s_tokenCounter++;
}
```

I just also discovered the mood does not change even if i use 'cast'

I am welcoming any tip on how i can get it to flip. Thanks :)

**Comment by theirrationalone** [comment]
Maybe, You're missing something in your moodnft contract..... 

Please verify those `if checks` in flipmood function, You're implementing them right... 

Check if you're using either`==` Or `=` inside `if checks`

**Comment by 78carla** [comment]
Hi, 
I have the same issue as @shittuisrael004.
(I wrote the interaction code scripts for the MintMoodNft and the FlipMoodNft. The mint works fine. The flipMoodNft runs successfully without any error but it does not really change the mood).
I have checked the flipmood function in my moodNft contract and it seems ok. 

My flipmood function in my moodNft contract is:
 
```
 function flipMood(uint256 tokenId) public {
        //Only want the owner to be able to flip the mood
        if (!_isApprovedOrOwner(_msgSender(), tokenId)) {
            revert MoodNft__CantFlipMoodIfNotOwner();
        }
        if (s_tokenIdToMood[tokenId] == Mood.HAPPY) {
            s_tokenIdToMood[tokenId] = Mood.SAD;
        } else {
            s_tokenIdToMood[tokenId] = Mood.HAPPY;
        }
    }
``` 

Here is my FlipMoodNft interactions code:

```
contract FlipMoodNft is Script {
    uint256 public constant TOKEN_ID_TO_FLIP = 0;

    function run() external {
        address mostRecentlyMoodDeployed = DevOpsTools
            .get_most_recent_deployment("MoodNft", block.chainid);
        flipNftOnContract(mostRecentlyMoodDeployed);
    }

    function flipNftOnContract(address moodContracAddress) public {
        vm.startBroadcast();
        MoodNft(moodContracAddress).flipMood(TOKEN_ID_TO_FLIP);
        vm.stopBroadcast();
    }
}
```

I also discovered the mood does not change even if I use directly the flipMood function on the verified contract on sepolia.etherscan.. 
Any suggestion for me? 
Thanks in advance,

**Reply by theirrationalone** [reply]
Heyy @78carla !, Your `flipMood` function and `interactions - flipMood call` seem Perfect. It would be Great If You could share your repo here. I'd like to test it to identify potential issues.

**Reply by 78carla** [reply]
Hi, 
I found the solution:

- on Metamask: I have deleted the minted NFT and I have added it again after the flip
- on https://testnets.opensea.io/: I have refreshed the metadata (refresh metadata option under the three dots on the top right).

All works fine...
